### ISSUE TITLE
Number field step validation

### ISSUE THREAD

**Issue by Kuuak** [wpmetabox/meta-box#921]
Hello there,

Unfortunately the answer on issue #916 isn't correct.
Indeed the step argument on number input was working fine until I updated the plugin from 4.8.5 to 4.8.7. Therefore it isn't the browsers which don't correctly validate, but the plugin who don't handle it well any more.

I reverted back to 4.8.5 (to be sure), and it works, then updated to 4.8.6 which introduce the issue.

I tested on both OS X 10.11.5 and Windows 10 with Firefox (47), Chrome (51), Safari (9.1.1), IE (11), Edge (20) and all browser react the same.

Here my meta definition:

``` php
array( 
    'name'  => __("Diamètre/Largeur", 'bvt-domain' ),
    'id'        => $prefix . '_diameter',
    'type'  => 'number',
    'desc'  => __( 'mm', 'bvt-domain' ),
    'step'  => 0.01,
),
```

Furthermore, the error message are exactly the same (text & style) for every browser which demonstrate the it is not the browser who handle the validation. 

Below the difference between the plugin error validation and the native Firefox HTML5 form validation.

**Metabox**
![metabox-error-validation](https://cloud.githubusercontent.com/assets/6989959/16730997/2e17b816-4776-11e6-8073-6322e29892ff.png)

**Firefox**
![firefox-html5-form-validation](https://cloud.githubusercontent.com/assets/6989959/16731024/4c4696b8-4776-11e6-8d79-746340ee21b3.jpg)

Thanks

**Comment by rilwis** [comment]
Thanks for reporting!

After checking with the plugin, I found that the JavaScript validation library causes the bug.

The check in the JS library seems logical here:

https://github.com/jzaefferer/jquery-validation/blob/master/src/core.js#L1403

However, on Firefox, I made a [small test](http://jsbin.com/vopuwetawu/edit?js,console) to see how it handles float point numbers and the result is very frustrating:

```
0.03 % 0.01 === 0: false
0.04 % 0.01 === 0: true
0.05 % 0.01 === 0: false
0.06 % 0.01 === 0: false
```

 I will find a way to exclude validation on fields when it’s not specified.
